File: examples/python/best-practices/avoid-vague-names.py
@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> dict:
    # Generic names don't help with debugging
    user = context.step(fetch_user(event["user_id"]), name="step1")
    validated = context.step(validate_user(user), name="step2")
    notification = context.step(send_notification(user), name="step3")
    return {"status": "completed"}

File: examples/java/best-practices/avoid-vague-names.java
// Generic names don't help with debugging
var user = ctx.step("step1", Map.class, stepCtx -> fetchUser(input.get("user_id")));
var validated = ctx.step("step2", Map.class, stepCtx -> validateUser(user));
var notification = ctx.step("step3", Map.class, stepCtx -> sendNotification(user));
return Map.of("status", "completed");
